problem81.rb

```diff
@@ -0,0 +1,22 @@
+#Create a function that takes a string (morse code) as an argument and returns an unencrypted string.
+#Examples
+#decodeMorse(".... . .-.. .--. -- . -.-.--") ➞ "HELP ME !"
+#decodeMorse("-.-. .... .- .-.. .-.. . -. --. .") ➞ "CHALLENGE"
+#decodeMorse(". -.. .- -... -... .. - -.-. .... .- .-.. .-.. . -. --. .") ➞ "EDABBIT CHALLENGE"
+def decodeMorse(str)
+  morseToDots = {
+    ".-" => "A", "-..." => "B", "-.-." => "C", "-.." => "D", "." => "E", "..-." => "F",
+    "--." => "G", "...." => "H", ".." => "I", ".---" => "J", "-.-" => "K", ".-.." => "L",
+    "--" => "M", "-." => "N", "---" => "O", ".--." => "P", "--.-" => "Q", ".-." => "R",
+    "..." => "S", "-" => "T", "..-" => "U", "...-" => "V", ".--" => "W", "-..-" => "X",
+    "-.--" => "Y", "--.." => "Z", "-----" => "0", ".----" => "1", "..---" => "2",
+    "...--" => "3", "....-" => "4", "....." => "5", "-...." => "6", "--..." => "7",
+    "---.." => "8", "----." => "9", "-.-.--" => "!", " " => " ", "..--.." => "?",
+    ".-.-.-" => ".", ".----." => " ", "---..." => " => ", "--..--" => " , ",
+  }
+  str.split("   ").map { |x| x.split(" ").map { |a| morseToDots[a] }.join }.join(" ")
```

Comment:
Are you sure this is more suitable solution for this challenge? Because you can check the complexity is n^3 - n^2, so try to provide more optimize solution if you want any help so let me know.